tests/test_distribuited/test_nats.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+import socketserver
+import concurrent.futures
+import subprocess
+
+from dffml.util.asynctestcase import AsyncTestCase
+
+
+from dffml.distribuited.orchestrator import NatsWorkerNode, NatsWorkerNodeConfig
+from dffml import DataFlow
+from dffml_feature_git.feature.operations import (
+    check_if_valid_git_repository_URL,
+)
+from dffml.operation.output import GetSingle
+from dffml.operation.db import (
+    DatabaseQueryConfig,
+    db_query_create_table,
+)
+
+import os
+import tempfile
+import asyncio
+
+from dffml.distribuited.orchestrator import (
+    NatsOrchestratorNode,
+    NatsOrchestratorNodeConfig,
+    NatsWorkerNode,
+    NatsWorkerNodeConfig,
+)
+from dffml import DataFlow
+from dffml_feature_git.feature.operations import (
+    check_if_valid_git_repository_URL,
+)
+from dffml.operation.output import GetSingle
+from dffml.db.sqlite import SqliteDatabase, SqliteDatabaseConfig
+from dffml.operation.db import (
+    DatabaseQueryConfig,
+    db_query_create_table,
+)
+
+
+class TestNatsOrchestrator(AsyncTestCase):
+    async def setUp(self):
+        self.nats_proc = subprocess.Popen(
+            ["nats-server", "-p", "-1",],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            start_new_session=True,
+        )
+
+        ready = False
+        listen_text = "Listening for client connections on "
+        while not ready:
+            line = self.nats_proc.stdout.readline().decode()
+            if listen_text in line:
+                line = line.strip().split(listen_text)
+                self.server_addr = line[-1]
+            if "Server is ready" in line:
+                ready = True
+
+    async def tearDown(self):
+        self.nats_proc.terminate()
+
+    async def test_run(self):
+        server = self.server_addr
+        subnode_1 = NatsWorkerNode(
```

Comment:
Let's instantiate 3 workers (configurable to N workers, maybe use a loop or something) and make sure that it still work.

Let's give each a different set of operations and make some of them overlap and some unique and makes sure that circular queues are setup appropriately.